Archipelago.cpp

```diff
@@ -241,34 +241,46 @@ void AP_SetClientVersion(AP_NetworkVersion* version) {
 }
 
 void AP_SendItem(int64_t idx) {
-    printf("AP: Checked '%s'. Informing Archipelago...\n", getLocationName(ap_game,idx).c_str());
+    AP_SendItem(std::vector<int64_t>{ idx });
+}
+void AP_SendItem(std::vector<int64_t> const& locations) {
+    printf("AP: ");
+    for (int64_t idx : locations) {
+        printf("Checked '%s'.", getLocationName(ap_game, idx).c_str());
+    }
+    printf("Informing Archipelago...\n", locations.size());
     if (multiworld) {
         Json::Value req_t;
         req_t[0]["cmd"] = "LocationChecks";
-        req_t[0]["locations"][0] = idx;
+        req_t[0]["locations"] = Json::arrayValue;
+        for (int64_t loc : locations) {
+            req_t[0]["locations"].append(loc);
+        };
         APSend(writer.write(req_t));
     } else {
-        for (auto itr : sp_save_root["checked_locations"]) {
-            if (itr.asInt64() == idx) {
-                return;
+        for (int64_t idx : locations) {
+            for (auto itr : sp_save_root["checked_locations"]) {
+                if (itr.asInt64() == idx) {
+                    return;
```

Comment:
Logic here seems broken. It reads to me as: If any location was checked before (in `sp_save_root`), do not update any.
It should probably instead skip those that are checked, and only consider the new ones.
You'll need to remove the entries in `sp_save_root` from `locations` and use that to create the fake message